ios/browser/ui/webui/ai_chat/ai_chat_ui.mm

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/ui/webui/ai_chat/ai_chat_ui.h"
+
+#include <memory>
+#include <string_view>
+#include <utility>
+
+#include "base/functional/bind.h"
+#include "base/notreached.h"
+#include "base/strings/sys_string_conversions.h"
+#include "brave/components/ai_chat/core/browser/ai_chat_service.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
+#include "brave/components/ai_chat/core/common/mojom/tab_tracker.mojom.h"
+#include "brave/components/ai_chat/core/common/mojom/untrusted_frame.mojom.h"
+#include "brave/components/ai_chat/resources/grit/ai_chat_ui_generated_map.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/ios/browser/ai_chat/ai_chat_service_factory.h"
+#include "brave/ios/browser/ui/webui/ai_chat/ai_chat_ui_page_handler.h"
+#include "brave/ios/browser/ui/webui/favicon_source.h"
+#include "brave/ios/web/webui/brave_web_ui_ios_data_source.h"
+#include "brave/ios/web/webui/brave_webui_utils.h"
+#include "components/favicon_base/favicon_url_parser.h"
+#include "components/grit/brave_components_resources.h"
+#include "components/grit/brave_components_webui_strings.h"
+#include "ios/chrome/browser/shared/model/profile/profile_ios.h"
+#include "ios/web/public/web_state.h"
+#include "ios/web/public/webui/url_data_source_ios.h"
+#include "ios/web/public/webui/web_ui_ios.h"
+#include "ios/web/public/webui/web_ui_ios_data_source.h"
+#include "ios/web/public/webui/web_ui_ios_message_handler.h"
+#include "third_party/abseil-cpp/absl/strings/str_format.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/webui/resource_path.h"
+#include "ui/base/webui/web_ui_util.h"
+
+AIChatUI::AIChatUI(web::WebUIIOS* web_ui, const GURL& url)
+    : web::WebUIIOSController(web_ui, url.host()),
+      profile_(ProfileIOS::FromWebUIIOS(web_ui)) {
+  DCHECK(profile_);
+  DCHECK(!profile_->IsOffTheRecord());
+
+  // Create a URLDataSource and add resources.
+  BraveWebUIIOSDataSource* source = brave::CreateAndAddWebUIDataSource(
+      web_ui, url.host(), kAiChatUiGenerated, IDR_AI_CHAT_UI_HTML);
+
+  source->AddResourcePath("styles.css", IDR_AI_CHAT_UI_CSS);
+  source->AddResourcePath("manifest.webmanifest", IDR_AI_CHAT_UI_MANIFEST);
+  source->AddResourcePath("pwa_icon.svg", IDR_AI_CHAT_UI_PWA_ICON);
+
+  source->AddLocalizedStrings(webui::kAiChatStrings);
+
+  constexpr bool kIsMobile = BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS);
+  source->AddBoolean("isMobile", kIsMobile);
```

Comment:
not sure why we didnt just compile in `true` here

Reply:
lol